nservicebus/best-practices.md

```diff
@@ -0,0 +1,130 @@
+
+In production, the creation of queues, database tables, and other tables should be scripted as a part of a DevOps process when the endpoint is installed. Methods to administer these resources differ for every message transport and persistence package. For many packages, Particular will provide a `dotnet` global tool, PowerShell library, or other means of performing operational scripting to create endpoints.
+
+
+:heavy_check_mark: **DO use the [Outbox feature](/nservicebus/outbox/) to provide consistency between business data modifications and messaging operations**
+
+Because message queues generally do not support any form of transactions, message handlers that modify business data in a database run into problems when the messages that are sent become inconsistent with the changes made to business data, resulting in ghost messages or phantom records.
+
+Using the Outbox guarantees exactly-once message processing by piggybacking on the local database transaction used to store business data.
+
+When using transports that only support [receive-only or sends-atomic-with-receives transactions](/transports/transactions.md), the Outbox ensures that outgoing messages are consistent with changes made to business data in the database.
+
+
+:x: **DO NOT use messaging for occasionally-connected clients**
+
+Message queues are long-lasting and durable. Occasionally-connected clients, such as mobile devices or client-side graphical user interfaces, should not receive messages from message queues. The durable nature of message queuing can result in a backlog of messages for a disconnected client which, if disconnected long enough, can result in queue quotas being exceeded. This can ultimately impact performance in other parts of the system.
+
+For occasionally-connected clients, consider another communication medium, such as in the [Near real-time transient clients sample](/samples/near-realtime-clients/) which communicates with clients using [SignalR](https://dotnet.microsoft.com/apps/aspnet/signalr).
+
+
+## System monitoring
+
+:heavy_check_mark: **DO install the [Heartbeats plugin](/monitoring/heartbeats/) in all endpoints to monitor for endpoint health**
+
+The Heartbeats plugin sends a message to [ServiceControl](/servicecontrol/) at regular intervals to demonstrate that the process is not only executing (which would be provided by any suite of system monitoring tools) but is also capable of interacting with the message transport.
+
+If ServiceControl stops receiving heartbeat messages from an endpoint, that endpoint will be shown as [inactive in ServicePulse](/monitoring/heartbeats/in-servicepulse.md). In addition, ServiceControl will publish a [`H`eartbeatStopped` event](/monitoring/heartbeats/notification-events.md) so that operations staff can be notified and respond.
```

Comment:
```suggestion
If ServiceControl stops receiving heartbeat messages from an endpoint, that endpoint will be shown as [inactive in ServicePulse](/monitoring/heartbeats/in-servicepulse.md). In addition, ServiceControl will publish a [`HeartbeatStopped` event](/monitoring/heartbeats/notification-events.md) so that operations staff can be notified and respond.
```